sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableServiceImpl.java

```diff
@@ -548,6 +566,7 @@ public CompletionStage<MutateRowResponse> writeRecord(KV<ByteString, Iterable<Mu
 
       CompletableFuture<MutateRowResponse> result = new CompletableFuture<>();
 
+      outstandingMutations += 1;
```

Comment:
I think so, but double-checking: safe to assume `writeRecord` won't happen concurrently? 

Otherwise, probably something like an AtomicInteger would guarantee accuracy

Reply:
thanks Bruno - yes, I was wondering that too, ha - since we create one per bundle, I don't think there's multiple concurrent calls to this logic.